salaTest/testshaperemove.cpp

```diff
@@ -0,0 +1,412 @@
+                std::find_if(connector.m_forward_segconns.begin(), connector.m_forward_segconns.end(),
+                             [&otherShapeRef](const std::pair<SegmentRef, int> &segmentRef) {
+                                 return segmentRef.first.ref == otherShapeRef;
+                             }) != connector.m_forward_segconns.end();
+            REQUIRE(inForConnections);
+        }
+        for (int otherShapeRef : expectedBackConnections) {
+            bool inBackConnections = std::find_if(connector.m_back_segconns.begin(), connector.m_back_segconns.end(),
+                                                  [&otherShapeRef](const std::pair<SegmentRef, int> &segmentRef) {
+                                                      return segmentRef.first.ref == otherShapeRef;
+                                                  }) != connector.m_back_segconns.end();
+            REQUIRE(inBackConnections);
+        }
+    }
+
+    SECTION("Delete from a segment map from the beginning") {
+        std::map<int, int> connectivitiesAfterRemoval;
+
+        for (auto iter = segmentMap->getAllShapes().begin(); iter != segmentMap->getAllShapes().end(); iter++) {
+            connectivitiesAfterRemoval[iter->first] =
+                segmentForConnections[iter->first].size() + segmentBackConnections[iter->first].size();
+        }
+
+        int shapeCount = segmentMap->getAllShapes().size();
+        for (size_t idx = shapeCount; idx > 0; idx--) {
+            int shapeRef = segmentMap->getAllShapes().begin()->first;
+            segmentMap->removeShape(shapeRef, false);
+            REQUIRE(segmentMap->getAllShapes().size() == idx - 1);
+            REQUIRE(segmentMap->getConnections().size() == idx - 1);
+
+            for (auto it = segmentMap->getAllShapes().begin(); it != segmentMap->getAllShapes().end(); it++) {
+                std::vector<int> connForConnections = segmentForConnections[it->first];
+                std::vector<int> connBackConnections = segmentBackConnections[it->first];
+                if (std::find(connForConnections.begin(), connForConnections.end(), shapeRef) !=
+                    connForConnections.end()) {
+                    // if the other shape contains this one, then remove from its connectivity
+                    // removal does not currently work for segment maps, so connectivity is not adjusted
+                    // connectivitiesAfterRemoval[it->first] = connectivitiesAfterRemoval[it->first] - 1;
+                } else if (std::find(connBackConnections.begin(), connBackConnections.end(), shapeRef) !=
+                           connBackConnections.end()) {
+                    // if the other shape contains this one, then remove from its connectivity
+                    // removal does not currently work for segment maps, so connectivity is not adjusted
+                    // connectivitiesAfterRemoval[it->first] = connectivitiesAfterRemoval[it->first] - 1;
+                }
+
```

Comment:
This looks like it can go?

Reply:
This code has been removed and will be created as a separate work-in-progress PR to fix segment removal